### PR TITLE
Implement Automatic API Version Management

### PR DESCRIPTION
**Pull request by ClayChipps**
A continuing headache that we deal with on the AEP projects is managing the upgrade of API versions with each release. In order to eliminate the manual effort associated with this, I have proposed a manual GitHub action which can be ran which will update all API Versions in the project and submit a PR for those changes. As of right now, there is no scheduling or automated runs, but at the minimum this should help reduce the manual effort.

- - -
This change is [Reviewable](https://reviewable.io/reviews/apex-enterprise-patterns/fflib-apex-mocks/141)

### CODE REVIEW

File: .github/workflows/upgrade.api.versions.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Bump Version
+        run: |
+          file_list=($(find ./sfdx-source -type f -exec grep -lP "<apiVersion>.*</apiVersion>" {} \;))
+          for file in "${file_list[@]}"; do
+            echo "Processing $file"
+            sed -i "s|<apiVersion>.*</apiVersion>|<apiVersion>${{inputs.api-version}}.0</apiVersion>|g" $file
+          done
+          sed -i "s|\"sourceApiVersion\": \".*\"|\"sourceApiVersion\": \"58.0\"|g" sfdx-project.json
```
Comment (stohn777) [inline .github/workflows/upgrade.api.versions.yml:24]: Should the "58.0" here be the "inputs.api-version" variable?

Comment (ctchipps) [reply .github/workflows/upgrade.api.versions.yml:24]: No :(

Comment (ctchipps) [reply .github/workflows/upgrade.api.versions.yml:24]: Yes :(

**Comment by ClayChipps** [comment]
@stohn777 I moved the bash script into an actual GitHub Action, can you take a look and let me know if this covers your concerns from before? :)

**Review by stohn777 — APPROVED** [review]

**Review by ImJohnMDaniel — APPROVED** [review]

**Review by ImJohnMDaniel — APPROVED** [review]

**Pull request merged**